Visitor/Arithmetic.php

```diff
@@ -245,16 +245,19 @@ public function visit(
 
             case 'token':
                 $value = $element->getValueValue();
+                $token = $element->getValueToken();
                 $out   = null;
 
-                if ('constant' === $element->getValueToken()) {
+                if ('constant' === $token) {
                     if (defined($value)) {
                         $out = constant($value);
                     } else {
                         $out = $this->getConstant($value);
                     }
-                } elseif ('id' === $element->getValueToken()) {
+                } elseif ('id' === $token) {
                     return $value;
+                } elseif ('binary' === $token || 'hex' === $token || 'octal' === $token) {
+                    $out = (float) eval('return ' . $value . ';');
```

Comment:
For binary: [`bindec`](http://php.net/bindec).
For hexa: [`hexdec`](http://php.net/hexdec).
For octal: [`octdec`](http://php.net/octdec).